pkg/controller/toolchainstatus/toolchainstatus_controller_test.go

```diff
@@ -636,6 +582,169 @@ func TestToolchainStatusConditions(t *testing.T) {
 		})
 	})
 }
+func TestToolchainStatusNotifications(t *testing.T) {
+	// set the operator name environment variable for all the tests which is used to get the host operator deployment name
+	restore := test.SetEnvVarsAndRestore(t, test.Env(k8sutil.OperatorNameEnvVar, defaultHostOperatorName))
+	defer restore()
+	requestName := configuration.DefaultToolchainStatusName
+
+	registrationService := newRegistrationServiceReady()
+	toolchainStatus := NewToolchainStatus()
+	memberStatus := newMemberStatusReady()
+	registrationServiceDeployment := newDeploymentWithConditions(registrationservice.ResourceName,
+		status.DeploymentAvailableCondition(), status.DeploymentProgressingCondition())
+
+	t.Run("Notification not created when host operator deployment ready", func(t *testing.T) {
+		// given
+		defer counter.Reset()
+		hostOperatorDeployment := newDeploymentWithConditions(defaultHostOperatorName,
+			status.DeploymentAvailableCondition())
+
+		toolchainStatus.Status.Conditions = []toolchainv1alpha1.Condition{
+			{
+				Type:               toolchainv1alpha1.ConditionReady,
+				Status:             corev1.ConditionTrue,
+				LastTransitionTime: metav1.Time{time.Now()},
+			},
+		}
+		defer func() { toolchainStatus.Status.Conditions = nil }()
+
+		reconciler, req, fakeClient := prepareReconcile(t, requestName, newResponseGood(),
+			newGetMemberClustersFuncReady, hostOperatorDeployment, memberStatus, registrationServiceDeployment,
+			registrationService, toolchainStatus)
+
+		// when
+		res, err := reconciler.Reconcile(req)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, requeueResult, res)
+
+		AssertThatToolchainStatus(t, req.Namespace, requestName, fakeClient).
+			HasCondition(componentsReady()).
+			HasHostOperatorStatus(hostOperatorStatusReady()).
+			HasMemberStatus(memberClusterSingleReady()).
+			HasRegistrationServiceStatus(registrationServiceReady())
+
+		// Lookup the notification
+		var notification toolchainv1alpha1.Notification
+		err = fakeClient.Get(context.Background(), test.NamespacedName(test.HostOperatorNs, "toolchainstatus-unready"),
+			&notification)
+		require.Error(t, err)
+		require.IsType(t, &errors.StatusError{}, err)
+		require.Equal(t, string(err.(*errors.StatusError).ErrStatus.Reason), "NotFound")
```

Comment:
It's `Equal(expected, actual)`. Not `Equal(actual, expected)`
Also you can use `k8s.io/apimachinery/pkg/api/errors.IsNotFound(err)`
```suggestion
		require.True(t, errors.IsNotFound(err))
```